src/main.rs

```diff
@@ -422,40 +393,85 @@ fn remote_server_options(matches: &ArgMatches) -> Result<(Option<Sources>, Optio
     Ok((sources, docker_config))
 }
 
-fn verification_options(
+fn verification_options(matches: &ArgMatches) -> Result<Option<VerificationSettings>> {
+    let verification_settings = if let Some(verification_config_path) =
```

Comment:
I think we should error out if the user is specifying the configuration time and, at the same time, also pub keys and annotations.

Reply:
True! will do.

Reply:
This can be done with:
```rust
    let verification_config = if let Some(verification_config_path) =
        matches.value_of("verification-config-path")
    {
        // verification-config has precedence over flags:
        if matches.is_present("verification-key") || matches.is_present("verification-annotation")
        {
            return Err(anyhow!(
                "verification-config-path flag has precedence over other verification flags"
            ));
        }
        Some(read_verification_file(Path::new(
            &verification_config_path,
        ))?)

```

But this is lossy. We will add new flags, and forget to add them to the list of flags that get checked for returning the error.

I would prefer to silently ignore flags, and just document that the config has precedence.

Reply:
From your next comment, giving higher precedence to cli flags from user instead of config, solves this.